### PR TITLE
write functional tests for finding elements by images

### PR DESCRIPTION
**Pull request by jlipps**
note that this depends on https://github.com/appium/appium-base-driver/pull/235 so it won't work until a new version of basedriver is published.

but i've run these tests locally with the new wd and basedriver linked, and they pass

### CODE REVIEW

File: test/functional/basic/find-e2e-specs.js [hunk 1]
```diff
@@ -400,4 +404,53 @@ describe('XCUITestDriver - find', function () {
...
+      el.value.should.match(/appium-image-element/);
+    });
+    it('should not find an image element that is not matched', async function () {
+      await driver.elementByImageFile(TESTAPP_BUTTON_IMG)
+        .should.eventually.be.rejectedWith(/Error response status: 7/);
+    });
+    it('should find anything with a threshold low enough', async function () {
+      const {imageMatchThreshold} = await driver.settings();
+      await driver.updateSettings({imageMatchThreshold: 0});
+      await driver.elementByImageFile(TESTAPP_BUTTON_IMG).should.eventually.exist;
```
Comment (imurchie) [inline test/functional/basic/find-e2e-specs.js]: If, for any reason, this fails, the image match threshold will remain 0. Perhaps better to put this in a try/finally block, if not a separate describe block with a before/after.

File: test/functional/basic/find-e2e-specs.js [hunk 2]
```diff
@@ -400,4 +404,53 @@ describe('XCUITestDriver - find', function () {
...
+      await driver.updateSettings({imageMatchThreshold});
+    });
+    it('should be able to get basic element properties', async function () {
+      let el = await driver.elementByImageFile(UICATALOG_LIST_IMG);
+      await el.isDisplayed().should.eventually.be.true;
+      let size = await el.getSize();
+      size.width.should.be.above(0);
+      size.height.should.be.above(0);
+      let loc = await el.getLocation();
+      (loc.x >= 0).should.be.true;
```
Comment (imurchie) [inline test/functional/basic/find-e2e-specs.js]: Does `loc.x.should.be.above(0);` not work? Seems strange to have two entirely different ways of doing the same thing in one test.

Comment (jlipps) [reply test/functional/basic/find-e2e-specs.js]: well I didn't want it to be above 0, i wanted it to be greater than or equal to zero, and i was too lazy to look up the chai docs to see how to make that work.

Comment (jlipps) [reply test/functional/basic/find-e2e-specs.js]: but I just did and it is `be.at.least`, so i will change it

File: test/functional/basic/find-e2e-specs.js (hunk 1 above)
Comment (jlipps) [reply test/functional/basic/find-e2e-specs.js]: mmmm, good point

**Comment by jlipps** [comment]
@imurchie ah yes. it looks like we need opencv4nodejs on travis for this to work. is that something we've done elsewhere?

**Comment by jlipps** [comment]
hmm yes, but only for linux. will have to explore more.

**Comment by jlipps** [comment]
yay, opencv-age!

**Comment by mykola-mokhnach** [comment]
@jlipps @imurchie do we still want to marge this PR?

**Comment by dpgraham** [comment]
This seems to have been forgotten about.... I rebased it and pushed it again and the build seems to be broken.

**Comment by dpgraham** [comment]
```Original error: opencv4nodejs module is required to use OpenCV features. Please install it first (npm i -g opencv4nodejs) and restart Appium.```

**Comment by dpgraham** [comment]
@jlipps sorry, I think my rebase may have messed some things up.

**Comment by dpgraham** [comment]
Closing. Can re-open if this gets picked up again.

**Pull request closed without merging**